src/main/java/com/google/sps/algorithms/AbuseDetection.java

```diff
@@ -61,29 +61,25 @@ public boolean addRequest(LocalTime currentTime) {
 
             returnValue = true;
             requestCounter++;
-            System.out.println("Current num: " + requestCounter);
 
         }
         else {
-            System.out.println("ELSE) Current num: " + requestCounter);
 
             for(Date currentDate : timesOfRequests) {
                 Instant instant = Instant.ofEpochMilli(currentDate.getTime());
                 LocalTime currentDateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault()).toLocalTime();
 
                 Duration timeDifference = Duration.between(currentDateTime, currentTime);
-                long difference = timeDifference.getSeconds();
-                System.out.println("The difference: " + difference);
-               
-                long initialTimePeriod = timePeriod.getSeconds();
-                System.out.println("The initalTimeperiod: " + initialTimePeriod);
+                int difference = timeDifference.compareTo(timePeriod);
 
+                long timePeriodlong = timePeriod.getSeconds();
+                int timePeriodInt = (int)timePeriodlong;
                
-                if(difference >= initialTimePeriod) {
+                if(difference >= timePeriodInt) {
```

Comment:
What does this comparison mean?

Reply:
Um it compares the difference between timperiod and timedifference in order to determine what gets removed and what gets added.

Reply:
So if the difference is greater than the timePeriod then that means that the time on the loop can be removed, and currentime that is passed in can be added in.

Reply:
Follow up question: what are the values of difference and timePeriodInt? What do these values represent?

Reply:
difference = compareTo of the initialTimePeriod Duration and the timeDifference 

timePeriod Int is the timePeriod duration converted down to an int

Reply:
compareTo returns 

- a positive integer if timeDifference > timePeriod
- a negative integer if timeDifference < timePeriod
- 0 if timeDifference == timePeriod

Is it meaningful to compare the result of compareTo with a timePeriod converted to seconds?

Reply:
I'm comparing mainly to handle the cases where the time request can get added, and one of the localtimes gets removed.

Reply:
I'm unsure what conditional would look like to add / remove the times

Reply:
timeDifference.compareTo(timePeriod) > 0 should be sufficient for comparing the timeDifference with the timePeriod

Reply:
Ok

Reply:
Implemented this.